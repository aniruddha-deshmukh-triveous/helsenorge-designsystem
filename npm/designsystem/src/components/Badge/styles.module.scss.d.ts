export type Styles = {
  badge: string;
  'badge--banana': string;
  'badge--blueberry': string;
  'badge--cherry': string;
  'badge--kiwi': string;
  'badge--neutral': string;
  'badge--oversized': string;
  'badge--plum': string;
  'badge--white': string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
