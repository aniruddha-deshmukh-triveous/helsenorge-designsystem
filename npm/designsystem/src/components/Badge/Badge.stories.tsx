import React from 'react';

import { ComponentStory, ComponentMeta } from '@storybook/react';

import { allPaletteNames } from '../../../.storybook/knobs';
import Badge from './Badge';

export default {
  title: 'Badge',
  component: Badge,
  argTypes: {
    children: {
      control: 'text',
      defaultValue: '1',
    },
    color: {
      control: 'select',
      options: allPaletteNames,
      defaultValue: 'blueberry',
    },
  },
} as ComponentMeta<typeof Badge>;

export const Default: ComponentStory<typeof Badge> = (args: any) => <Badge {...args}>{args.children}</Badge>;
