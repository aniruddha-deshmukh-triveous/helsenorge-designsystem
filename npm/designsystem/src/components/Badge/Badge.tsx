import React from 'react';
import classNames from 'classnames';

import { PaletteNames } from '../../theme/palette';

import badgeStyles from './styles.module.scss';

export type BadgeColors = PaletteNames;
export type BadgeChildren = string | number;

interface BadgeProps {
  /** Sets the content of the badge. */
  children: BadgeChildren;
  /** Adds custom classes to the element. */
  className?: string;
  /** Changes the badge background color. */
  color?: BadgeColors;
  /** Sets the data-testid attribute. */
  testId?: string;
}

const Badge = React.forwardRef(function BadgeForwardedRef(props: BadgeProps, ref: React.ForwardedRef<HTMLElement>) {
  const { children, className = '', color = 'black', testId } = props;
  const oversized = children.toString().length > 2;
  const badgeClasses = classNames(
    badgeStyles.badge,
    {
      [badgeStyles['badge--oversized']]: oversized,
      [badgeStyles['badge--black']]: color === 'black',
      [badgeStyles['badge--white']]: color === 'white',
      [badgeStyles['badge--blueberry']]: color === 'blueberry',
      [badgeStyles['badge--banana']]: color === 'banana',
      [badgeStyles['badge--cherry']]: color === 'cherry',
      [badgeStyles['badge--kiwi']]: color === 'kiwi',
      [badgeStyles['badge--neutral']]: color === 'neutral',
      [badgeStyles['badge--plum']]: color === 'plum',
    },
    className
  );

  return (
    <span className={badgeClasses} ref={ref} data-testid={testId}>
      {children}
    </span>
  );
});

export default Badge;
