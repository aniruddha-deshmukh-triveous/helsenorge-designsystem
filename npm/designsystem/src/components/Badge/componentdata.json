{
  "props": {
    "children": {
      "defaultValue": null,
      "description": "Sets the content of the badge.",
      "name": "children",
      "parent": {
        "fileName": "src/components/Badge/Badge.tsx",
        "name": "BadgeProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Badge/Badge.tsx",
          "name": "BadgeProps"
        }
      ],
      "required": true,
      "type": {
        "name": "ReactText"
      }
    },
    "className": {
      "defaultValue": null,
      "description": "Adds custom classes to the element.",
      "name": "className",
      "parent": {
        "fileName": "src/components/Badge/Badge.tsx",
        "name": "BadgeProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Badge/Badge.tsx",
          "name": "BadgeProps"
        }
      ],
      "required": false,
      "type": {
        "name": "string"
      }
    },
    "color": {
      "defaultValue": null,
      "description": "Changes the badge background color.",
      "name": "color",
      "parent": {
        "fileName": "src/components/Badge/Badge.tsx",
        "name": "BadgeProps"
      },
      "declarations": [
        {
          "fileName": "src/components/Badge/Badge.tsx",
          "name": "BadgeProps"
        }
      ],
      "required": false,
      "type": {
        "name": "enum",
        "raw": "\"banana\" | \"blueberry\" | \"cherry\" | \"kiwi\" | \"neutral\" | \"plum\" | \"black\" | \"white\" | undefined",
        "value": [
          {
            "value": "undefined"
          },
          {
            "value": "\"banana\""
          },
          {
            "value": "\"blueberry\""
          },
          {
            "value": "\"cherry\""
          },
          {
            "value": "\"kiwi\""
          },
          {
            "value": "\"neutral\""
          },
          {
            "value": "\"plum\""
          },
          {
            "value": "\"black\""
          },
          {
            "value": "\"white\""
          }
        ]
      }
    }
  }
}
