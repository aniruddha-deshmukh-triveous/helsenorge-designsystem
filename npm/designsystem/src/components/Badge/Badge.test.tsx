import React from 'react';
import { render, screen } from '@testing-library/react';
import Badge from './Badge';

const badgeNumber = 123;

test('displays text', (): void => {
  const { container, getByText } = render(<Badge>{badgeNumber}</Badge>);
  expect(getByText(badgeNumber.toString())).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

/* 
  TODO: Waiting for issues regarding babel and types for this to be implemented
  Seems to be version issues because of v5 release.
*/
// test('has the right color', (): void => {
//   const {container} = render(<Badge color="blood200">{badgeNumber}</Badge>);
//   expect(container).toMatchSnapshot();
// });

describe('Gitt at Badge skal vises', (): void => {
  describe('Når testId-prop er satt', (): void => {
    test('Så kan komponenten finnes ved hjelp av testId', (): void => {
      render(<Badge testId="bare-tester">123</Badge>);

      const component = screen.getByTestId('bare-tester');
      expect(component).toBeVisible();
    });
  });
});
