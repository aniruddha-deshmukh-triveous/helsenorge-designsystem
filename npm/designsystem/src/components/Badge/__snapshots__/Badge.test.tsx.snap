// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Gitt at Badge skal vises > Når Badge har tekst > Så vises teksten 1`] = `
<div>
  <span
    class="badge"
    data-analyticsid="badge"
  >
    123
  </span>
</div>
`;
